### PR TITLE
MINIFICPP-595: Provide basic support for windows.

### PR DESCRIPTION
**Pull request by phrocker**
MINIFICPP-32: Add windows event log reader
MINIFICPP-596: Build core and libminifi artifacts. Must abstract
features that are operating system dependent, such as uuid

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: CMakeSettings.json [hunk 1]
```diff
@@ -0,0 +1,146 @@
+{
+  // See https://go.microsoft.com//fwlink//?linkid=834763 for more information about this file.
+  "configurations": [
+    {
+      "name": "x86-Debug",
```
Comment (apiri) [inline CMakeSettings.json]: Should we remove these other configurations (those besides x64-Debug) until they are functional?

File: LICENSE [hunk 2]
```diff
@@ -1427,3 +1451,183 @@ This product bundles 'bsdiff' which is available under a "2-clause BSD" license.
 	 STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 	 IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 	 POSSIBILITY OF SUCH DAMAGE.
+
+ This product bundles 'Simple-Windows-Posix-Semaphore' which is available under an Apache v2 License.
```
Comment (apiri) [inline LICENSE]: As this is ALv2 it shouldn't be needed in the LICENSE.  There does not appear to be a NOTICE which would have to get incorporated.

**Comment by phrocker** [comment]
@apiri it is not anticipated that OpenSSL be installed for this release. Will check. Thanks!

File: LICENSE (hunk 2 above)
Comment (phrocker) [reply LICENSE]: I always felt this was good hygiene as a practice. Further I always thought that Category A licenses needed to adhere to the attribution portion of the aforementioned license. 4.d of http://apache.org/licenses/LICENSE-2.0 led me to believe this is a smart inclusion. I assume there is other language that makes this recursive attribution assumed?

Comment (apiri) [reply LICENSE]: The guidance from ASF is here http://www.apache.org/dev/licensing-howto.html#alv2-dep.  To your point, and in light of the ASF guidance, I believe we could just list the project and its version omitting the full license text?

Comment (phrocker) [reply LICENSE]: @apiri I added this very early on into the branch and thought I only had the first line ( also coincidentally only saw this in your comment ). Yes, totally agree that all of that text below the project and version is completely unnecessary. thanks!

**Comment by apiri** [comment]
Had some issues on a VM environment with build but seems to be just something on that system's configuration.  A clean Windows 10 VM as well as a baremetal Win 10 built and worked without issue.  Verified some simple flows and expected operation.  Will get these items merged.

**Pull request closed without merging**